spec/validations/source_spec.rb

```diff
@@ -1,139 +1,47 @@
+      expect(package).to receive(:manifest).and_return(requirements_only_manifest)
     end
 
-    context 'when the package includes app.js' do
-      before do
-        allow(package).to receive(:js_files) { files }
-      end
+    it 'returns error if package contains source files' do
+      package.js_files << 'random_js_file.js'
 
-      it 'should have an error' do
-        errors = ZendeskAppsSupport::Validations::Source.call(package)
-        expect(errors.first.key).to eql :no_code_for_ifo_notemplate
-      end
+      expect(validation_error).to receive(:new).with(:no_code_for_ifo_notemplate)
     end
 
-    context 'when the package includes a library' do
-      before do
-        allow(package).to receive(:js_files) { [double('AppFile', relative_path: 'lib/slapp.js')] }
-      end
-
-      it 'should have an error' do
-        errors = ZendeskAppsSupport::Validations::Source.call(package)
-        expect(errors.first.key).to eql :no_code_for_ifo_notemplate
-      end
+    it 'returns nothing if package has no source files' do
+      expect(validation_error).not_to receive(:new).with(:no_code_for_ifo_notemplate)
     end
+  end
 
-    context 'when the package includes app.css' do
-      before do
-        allow(package).to receive(:js_files) { [] }
-        allow(package).to receive(:app_css) { 'div {display: none;}' }
-      end
-
-      it 'should have an error' do
-        errors = ZendeskAppsSupport::Validations::Source.call(package)
-        expect(errors.first.key).to eql :no_code_for_ifo_notemplate
-      end
+  context 'when validating marketing_only app' do
+    before do
+      marketing_only_manifest =
+        ZendeskAppsSupport::Manifest.new({ requirementsOnly: false, marketingOnly: true }.to_json)
+      expect(package).to receive(:manifest).and_return(marketing_only_manifest)
     end
 
-    context 'when the package includes a template' do
-      before do
-        allow(package).to receive(:js_files) { [] }
-        allow(package).to receive(:app_css) { '' }
-        allow(package).to receive(:template_files) { [ 'templates/layout.hdbs' ] }
-      end
+    it 'returns error if package contains source files' do
+      package.template_files << 'random_template.html'
 
-      it 'should have an error' do
-        errors = ZendeskAppsSupport::Validations::Source.call(package)
-        expect(errors.first.key).to eql :no_code_for_ifo_notemplate
-      end
+      expect(validation_error).to receive(:new).with(:no_code_for_ifo_notemplate)
     end
-  end
 
-  context 'when not requirements only' do
-    it 'should have an error when app.js is missing' do
-      errors = ZendeskAppsSupport::Validations::Source.call(package)
-      expect(errors.first.to_s).to eql 'Could not find app.js'
+    it 'returns nothing if package has no source files' do
+      expect(validation_error).not_to receive(:new).with(:no_code_for_ifo_notemplate)
```

Comment:
Same suggestion as https://github.com/zendesk/zendesk_apps_support/pull/234/files#r300911795

Reply:
Actually this test is duplicate now. Perhaps it can be removed? Otherwise `package.template_files = []` first to differentiate it?

Reply:
As discussed in person, js_files and template_files in package is defined on [this line](https://github.com/zendesk/zendesk_apps_support/blob/ce268a70cbefba0276d7dcaf7558aa5b37ad114b/spec/validations/source_spec.rb#L8)

```
line 8
let(:package) { double('Package', js_files: [], template_files: [], app_css: '') }
```